remapping_commands.sh

```diff
@@ -0,0 +1,181 @@
+echo "#### Building index: may take a while ####"
+bowtie2-build $newgenome index
+echo "#### Mapping results: ####"
+bowtie2 -f -x index variant_reads.out.fa | samtools view -bS - > reads_aligned.bam
+#example output:
+#>1094 reads; of these:
+#>  1094 (100.00%) were unpaired; of these:
+#>   7 (0.64%) aligned 0 times
+#>    48 (4.39%) aligned exactly 1 time
+#>    1039 (94.97%) aligned >1 times
+#>99.36% overall alignment rate
+
+
+
+
+
+#3) Data extraction
+echo '------------------------------3) Data extraction-----------------------------'
+#extract chromosome, rsID, position on new genome, original variant allele, qual, filter and info fields
+samtools sort reads_aligned.bam -o reads_aligned.sorted.bam
+samtools index reads_aligned.sorted.bam
+./reverse_strand.py -i reads_aligned.sorted.bam -o variants_remapped.vcf
+rm reads_aligned.bam reads_aligned.sorted.bam reads_aligned.sorted.bam.bai
+
+#add interval for variant position in bed format (required by getfasta)
+awk '{print $1"\t"$2-1"\t"$2"\t"$3"\t"$4"\t"$5"\t"$6"\t"$7}' variants_remapped.vcf > variants_remapped.bed
+
+#getfasta to get the REF genome alleles and then extract the genome alleles without the positions
+bedtools getfasta -tab -fi $newgenome -bed variants_remapped.bed | awk '{print $2}'> genome_alleles.txt
+rm variants_remapped.bed
+#make all alleles caps (sometimes there are lower-case bases)
+tr a-z A-Z < genome_alleles.txt > genome_alleles.fixed.txt
+
+#paste all the information into the correct columns
+awk '{print $1"\t"$2"\t"$3}' variants_remapped.vcf > temp_first_3_columns.txt
+awk '{print $4"\t"$5"\t"$6"\t"$7}' variants_remapped.vcf > temp_last_4_columns.txt
+paste temp_first_3_columns.txt genome_alleles.fixed.txt temp_last_4_columns.txt > var_pre_final.vcf
+rm temp_first_3_columns.txt temp_last_4_columns.txt genome_alleles.txt variants_remapped.vcf genome_alleles.fixed.txt
+
+
+#header:
+#create list of contigs/chromosomes to be added to the header
+awk '{print $1}' var_pre_final.vcf | sort | uniq > contig_names.txt
+while read CHR; do echo "##contig=<ID=$CHR>"; done < contig_names.txt > contigs.txt
+rm contig_names.txt
+
+#add the reference assembly
+echo "##reference="$newgenomeaccession >> contigs.txt
+
+#copy everything that isn't #CHROM (the column titles), ##contig or ##reference from the old header to a temp
+awk '($1 !~ /^##contig/ && $1 !~ /^##reference/ && $1 !~ /^#CHROM/) {print $0}' vcf_header.txt > temp_header.txt
+rm vcf_header.txt
+
+#add the two headers together and add the column names
+cat temp_header.txt contigs.txt > final_header.txt
+echo -e "#CHROM\tPOS\tID\tREF\tALT\tQUAL\tFILTER\tINFO" >> final_header.txt
+rm temp_header.txt contigs.txt
+
+#add header to the vcf file:
+cat final_header.txt | cat - var_pre_final.vcf > temp && mv temp $outfile
+rm final_header.txt var_pre_final.vcf
+
+
+echo "-----------------------------------Results-----------------------------------"
+echo "Total number of input variants:"
+grep "^[^#]" $vcffile | wc -l
```

Comment:
At least some of these statistics can be computed by `bcftools stats` instead
https://samtools.github.io/bcftools/bcftools.html#stats

Reply:
I have added this command to the pipeline, although I have left my results calculations as they are a quick way to get the numbers I need to fill in the spreadsheet.